passes/techmap/zinit.cc

```diff
@@ -139,11 +139,26 @@ struct ZinitPass : public Pass {
 
 				cell->setPort(ID(D), sig_d);
 				cell->setPort(ID(Q), initwire);
+
+				/*if (cell->type.in(ID($_DFFSR_NNN_), ID($_DFFSR_NNP_), ID($_DFFSR_NPN_), ID($_DFFSR_NPP_),
+							ID($_DFFSR_PNN_), ID($_DFFSR_PNP_), ID($_DFFSR_PPN_), ID($_DFFSR_PPP_)))
+				{
+					// TODO: I think I need a $_DFFRS_* cell where S has priority over R...
+					std::swap(cell->connections_.at(ID(R)), cell->connections_.at(ID(S)));
+				}
+				else*/ if (cell->type.in(ID($_DFF_NN0_), ID($_DFF_NN1_), ID($_DFF_NP0_), ID($_DFF_NP1_),
+							ID($_DFF_PN0_), ID($_DFF_PN1_), ID($_DFF_PP0_), ID($_DFF_PP1_)))
+				{
+					std::string t = cell->type.str();
+					t[8] = (t[8] == '0' ? '1' : '0');
+					cell->type = t;
+				}
 			}
 
-			for (auto &it : initbits)
-				if (donebits.count(it.first) == 0)
-					log_error("Failed to handle init bit %s = %s.\n", log_signal(it.first), log_signal(it.second));
+			if (!design->selected_whole_module(module))
```

Comment:
I think this should fail if we did not handle all init bits. When `zinit` is used then it's usually part of a flow where the later stages ignore init attributes and just assume everything is zero initialized. So not failing here will result in an incorrect output netlist for the purpose of whatever flow this is part of.

Reply:
I've noticed that there are a lot of `\init` attributes leftover, even after a full `synth_*` run. We will need to eliminate them all otherwise this pass won't be usable at all.

Reply:
Okay, I've elected to only erase the `(* init *)` bits that were actually consumed by a supported cell.